src/net.c

```diff
@@ -174,6 +174,69 @@ static int addattr_l(struct nlmsghdr *n, int maxlen, int type, void *data, int a
 	return 0;
 }
 
+static int wait_for_nic(const char *nic_name)
+{
+	struct sockaddr_nl nls;
+	char buf[512];
+	int fd = -1;
+	int ret = -1;
+	char path[512];
+
+	sprintf(path, "/sys/class/net/%s/ifindex", nic_name);
+
+	memset(&nls,0,sizeof(struct sockaddr_nl));
+	nls.nl_family = AF_NETLINK;
+	nls.nl_pid = getpid();
+	nls.nl_groups = -1;
+
+	fd = socket(PF_NETLINK, SOCK_DGRAM, NETLINK_KOBJECT_UEVENT);
+	if (fd==-1){
+		perror("Open socket failed\n");
+		return -1;
+	}
+
+	if ( access( path, F_OK ) != -1 ) {
```

Comment:
@sameo I think access needs to be called at two places, once could be called before the socket creation. (to avoid the socket creation and binding) and the other place it definitely needs to be called is  after the bind call to make sure no udev events are missed till bind() is called.